modules/codegen/src/smithy4s/codegen/SmithyResources.scala

```diff
@@ -0,0 +1,61 @@
+package smithy4s.codegen
+
+/**
+  * This construct aims at adding metadata information to the classpath to let Smithy4s
+  * know about code that may have been generated in upstream modules.
+  *
+  * A "smithy4sGenerated" metadata holds a list of all namespaces that will have already
+  * been generated by smithy4s.
+  */
+private[smithy4s] object SmithyResources {
+
+  def produce(
```

Comment:
I like when this kind of syncing logic, wipe the target folder and start from scratch. This version will leave leftover files behind if some namespace disappear from a module

Reply:
that's the status quo really, if you regenerate you may end up with leftover files that should no longer be generated

Reply:
It's problematic to wipe because other source/resource generators may exist and tap into the same directories (which some editors, like Intellij, happen to hardcode in their Scala support). That's why you have to return the list of files you've generated in source/resource generators, so that the build tool can ignore potential leftovers.